File: lambda/handler.rb
# frozen_string_literal: true

require 'ferrum'
require 'json'
require 'cgi'

def handler(event:, context:)
  html = CGI.unescape_html(event['html'])
  browser = Ferrum::Browser.new(browser_options: { 'no-sandbox': nil })
  browser.frames.first.set_content(html)
  base64 = browser.screenshot(full: true)
  { data: base64 }
end

File: lambda/handler.js
const puppeteer = require('puppeteer-core');
const he = require('he');

exports.handler =  async function(event, context) {
  const html = he.decode(event['html']);
  const browser = await puppeteer.connect({ browserWSEndpoint: `wss://chrome.browserless.io?token=${process.env.BROWSER_TOKEN}` });
  const page = await browser.newPage();

  await page.setContent(html);
  const base64 = await page.screenshot({ encoding: "base64", fullPage: true })

  await browser.close();

  return { base64 }
}
